lib/Polaris.Class.ps1

```diff
@@ -1,324 +1,319 @@
+	) {
+		if ($null -eq $Path) {
+			throw [ArgumentNullException]::new("path")
+		}
+		if ($null -eq $Method) {
+			throw [ArgumentNullException]::new("method")
+		}
+
+		[string]$SanitizedPath = $Path.TrimEnd('/')
+
+		if ([string]::IsNullOrEmpty($SanitizedPath)) { $SanitizedPath = "/" }
+
+		$this.ScriptBlockRoutes[$SanitizedPath].Remove($Method)
+		if ($this.ScriptBlockRoutes[$SanitizedPath].Count -eq 0) {
+			$this.ScriptBlockRoutes.Remove($SanitizedPath)
+		}
+	}
+
+	AddMiddleware (
+		[string]$Name,
+		[scriptblock]$Scriptblock
+	) {
+		if ($null -eq $Scriptblock) {
+			throw [ArgumentNullException]::new("scriptBlock")
+		}
+		$this.RouteMiddleware.Add([PolarisMiddleware]@{
+				'Name' = $Name
+				'ScriptBlock' = $Scriptblock
+			})
+	}
+
+	RemoveMiddleware ([string]$Name) {
+		if ($null -eq $Name) {
+			throw [ArgumentNullException]::new("name")
+		}
+		$this.RouteMiddleware.RemoveAll(
+			[Predicate[PolarisMiddleWare]]([scriptblock]::Create("`$args[0].Name -eq '$Name'"))
+		)
+	}
+
+	[void] Start (
+		[int]$Port = 3000
+	) {
+		$this.StopServer = $false
+		$this.InitListener($Port)
+		$this.Listener.BeginGetContext($this.ContextHandler,$this)
+		$this.Log("App listening on Port: " + $Port + "!")
+	}
+
+	[void] Stop () {
+		$this.StopServer = $true
+		$this.Listener.Close()
+		$this.Listener.Dispose()
+		$this.Log("Server Stopped.")
+	}
+
+	[void] InitListener ([int]$Port) {
+		$this.Port = $Port
+
+		$this.Listener = [System.Net.HttpListener]::new()
+
+		# If user is on a non-windows system or windows as administrator
+		if ([System.Environment]::OSVersion.Platform -ne [System.PlatformID]::Win32NT -or
+			([System.Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT -and
+				([System.Security.Principal.WindowsPrincipal]::new([System.Security.Principal.WindowsIdentity]::GetCurrent())).IsInRole([System.Security.Principal.WindowsBuiltInRole]::Administrator))) {
+			$this.Listener.Prefixes.Add("http://+:" + $this.Port + "/")
+		}
+		else {
+			$this.Listener.Prefixes.Add("http://localhost:" + $this.Port + "/")
+		}
+
+		$this.Listener.Start()
+	}
+
+    static [void] Send (
+        [System.Net.HttpListenerResponse]$RawResponse, 
+        [PolarisResponse]$Response
+	) {
         try {
-            [Polaris]::Send($rawResponse, $response.ByteResponse, $response.StatusCode, $response.ContentType, $response.Headers );
+            if ($Response.StreamResponse) {
+                [Polaris]::Send(
+                    $RawResponse,
+                    $Response.StreamResponse,
+                    $Response.StatusCode,
+                    $Response.ContentType,
+                    $Response.Headers
+                )
+            }
+            else {
+                [Polaris]::Send(
+					$RawResponse,
+					$Response.ByteResponse,
+					$Response.StatusCode,
+					$Response.ContentType,
+					$Response.Headers
+				)
+            }
         }
         catch {
-            throw $_         
+            throw $_
```

Comment:
same deal with the try/catch. Is this to convert it from a .NET exception to a PowerShell exception?